sql/core/src/main/scala/org/apache/spark/sql/execution/python/AttachDistributedSequenceExec.scala

```diff
@@ -40,15 +42,45 @@ case class AttachDistributedSequenceExec(
 
   override def outputPartitioning: Partitioning = child.outputPartitioning
 
+  @transient private var cached: RDD[InternalRow] = _
+
   override protected def doExecute(): RDD[InternalRow] = {
-    val childRDD = child.execute().map(_.copy())
-    val checkpointed = if (childRDD.getNumPartitions > 1) {
-      // to avoid execute multiple jobs. zipWithIndex launches a Spark job.
-      childRDD.localCheckpoint()
+    val childRDD = child.execute()
```

Comment:
Should we fallback to local checkpoint if AQE is disabled?

Reply:
I guess we should avoid using `checkpoint` even if AQE is disabled, just because it is unreliable: a local checkpointed rdd can not be recomputed if it is unfortunately evicted from memory, or due to something like dynamic allocation.

if we want to support checkpointing here, what about adding `localcheckpoint` and `checkpoint` as new options in this new configration?

Reply:
Hm, maybe let's just keep the legacy behaviour as is for now is AQE is disabled. Keeping the behaviour as is is fine but changing it to a new behaviour is a different story.

Reply:
ok

Reply:
maybe better to make `localcheckpoint` an option of `compute.distributed_sequence_index_storage_level`, so that we can fallback to the original behavior no matter whether AQE is enabled?

Reply:
SGTM